### PR TITLE
SOLR-15987: Upgrade slf4j to 1.7.35 and remove ant-compat/force-versions.gradle

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-15987

### CODE REVIEW

File: solr/licenses/slf4j-api-1.7.25.jar.sha1 [hunk 1]
```diff
@@ -0,0 +1 @@
+da76ca59f6a57ee3102f8f9bd9cee742973efa8a
```
Comment (madrob) [inline solr/licenses/slf4j-api-1.7.25.jar.sha1:1]: It's strange that we are getting 1.7.25 (here) and also 1.7.30 resolved...

Comment (risdenk) [reply solr/licenses/slf4j-api-1.7.25.jar.sha1:1]: Good catch - most likely transitive dependency resolution across all the modules. Its easy to force to 1.7.30 in versions.props. I didn't bump it but probably should.

**Comment by risdenk** [comment]
FWIW commons-logging will get cleaned up in SOLR-15992

**Review by dsmiley — COMMENTED** [review]
I think the upgrades you are making should be called out in JIRA.  Dependency changes should be more visible to the project overall (seen/noticed).  The means to which this is happening (removing force-versions.gradle which I didn't know about) is kind of secondary internal detail.

**Comment by risdenk** [comment]
@dsmiley yea thats fine. I wasn't planning on upgrading slf4j, but there is no way with the current build to keep a specific version - except with what is being removed here. There is really no good reason these versions are pegged to this. https://github.com/apache/solr/commits/main/gradle/ant-compat/force-versions.gradle just shows it was to "match the ant build". 

So the tl;dr is that this shouldn't be here (see comments about all these ant-compat files) and sure can call out more detail about the upgrade.

**Comment by dweiss** [comment]
> There is really no good reason these versions are pegged to this. https://github.com/apache/solr/commits/main/gradle/ant-compat/force-versions.gradle just shows it was to "match the ant build".

Yep, this was done to make the gradle build produce identical outputs as ant. Otherwise palantir's consistent versions would have picked the highest version of a dependency from those available in the dependency set.

**Review by dweiss — APPROVED** [review]
LGTM.

**Comment by risdenk** [comment]
Added solr/CHANGES.txt and updated the commit message to call out the slf4j upgrade.

**Pull request merged**